src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -0,0 +1,435 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final Gson serializer = new Gson();
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    private static final String JSON_EVENT_TYPE_ATTRIBUTE = "EventType";
+    private static final String JSON_NAME_ATTRIBUTE = "Name";
+    private static final String JSON_ATTRIBUTES_ATTRIBUTE = "TypeAttributes";
+    private static final String JSON_TIMESTAMP_ATTRIBUTE = "TimeStampAttributeName";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.get(EventTypeEndpoint.getEventTypesBaseUri(), this::getEventType);
+        sparkService.get(EventTypeEndpoint.getEventTypeBaseUri(), this::getEventTypes);
+        sparkService.post(EventTypeEndpoint.getCreateEventTypeBaseUri(), this::createEventType);
+        sparkService.delete(EventTypeEndpoint.getDeleteEventTypeBaseUri(), this::deleteEventType);
+        sparkService.get(EventTypeEndpoint.getEventTypeAttributesBaseUri(), this::getEventTypeAttributes);
+        sparkService.get(EventTypeEndpoint.getEventTypeQueriesBaseUri(), this::getEventTypeQueries);
+        sparkService.get(EventTypeEndpoint.getEventTypeEntityMappingsBaseUri(), this::getEventTypeEntityMappings);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypes(Request request, Response response) {
+        List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+        JsonArray jsonEventTypes = new JsonArray();
+        // TODO where does the key for each event type come from? ("EventType{...}")
+        for (EventType eventType : eventTypes) {
+            jsonEventTypes.add(getEventTypeJson(eventType));
+        }
+
+        return serializer.toJson(jsonEventTypes);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventType(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long eventTypeId = getEventTypeId(request);
+        EventType eventType = PersistenceAdapterImpl.getInstance().getEventType(eventTypeId);
+        JsonObject jsonEventTypes = getEventTypeJson(eventType);
+
+        return serializer.toJson(jsonEventTypes);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventType(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonEventType = jsonBody.get(JSON_EVENT_TYPE_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonEventType == null) {
+                halt(HttpStatusCodes.ERROR, "no event type given in body");

```

Comment:
Use `BAD_REQUEST` instead of `ERROR`, since this is not an internal server error, but an user error